src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1881,21 +1853,40 @@ void updateEscapeTo(BoundExpression argument, RefKind refKind, uint scopeOfTheCo
             };
         }
 
+#if DEBUG
+        private static bool AllParametersHaveArguments(
```

Comment:
It looks like we don't create arguments for params arrays during binding. https://github.com/dotnet/roslyn/issues/49602

Consider testing a case like:

```cs
using System;

Span<int> M1(Span<int> input, params object[] arr) { return input; }

Span<int> M2()
{
    return M1(stackalloc int[] { 1, 2, 3 });
}
```

We might want to adjust this step to ensure that all parameters *which may factor into escape analysis* have arguments. Or just specifically permit a parameter with `IsParams == true && Type.TypeKind == TypeKind.Array` here, since that's likely the only kind we will miss.

Reply:
Good catch, thanks.